src/threading.c

```diff
@@ -848,6 +848,138 @@ JL_DLLEXPORT int jl_alignment(size_t sz)
     return jl_gc_alignment(sz);
 }
 
+// Heartbeat mechanism for Julia's task scheduler
```

Comment:
This could probably go in its own .c file?

Reply:
It could, yes. That would require touching `Makefile`s so I decided to keep the patch small for now.

Reply:
Leaving a note to resolve this, we should consider reorganizing the code when we upstream this.